plugins/workspace-backpack/src/backpack_monkey_patch.js

```diff
@@ -0,0 +1,175 @@
+/**
+ * @license
+ * Copyright 2020 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @fileoverview A monkeypatch in Blockly to support a backpack.
+ * @author kozbial@google.com (Monica Kozbial)
+ */
+import * as Blockly from 'blockly/core';
+
+
+// TODO: No way to do this currently without monkeypatching Blockly.
+(() => {
+  /* eslint-disable no-import-assign */
+  /**
+   * Close tooltips, context menus, dropdown selections, etc.
+   * @param {boolean=} opt_allowToolbox If true, don't close the toolbox.
+   */
+  Blockly.hideChaff = function(opt_allowToolbox) {
+    Blockly.Tooltip.hide();
+    Blockly.WidgetDiv.hide();
+    Blockly.DropDownDiv.hideWithoutAnimation();
+    if (!opt_allowToolbox) {
+      const workspace = Blockly.getMainWorkspace();
+      // For now the trashcan flyout always autocloses because it overlays the
+      // trashcan UI (no trashcan to click to close it).
+      if (workspace.trashcan &&
+          workspace.trashcan.flyout) {
+        workspace.trashcan.closeFlyout();
+      }
+      const toolbox = workspace.getToolbox();
+      if (toolbox &&
+          toolbox.getFlyout() &&
+          toolbox.getFlyout().autoClose) {
+        toolbox.clearSelection();
+      }
+      if (workspace.backpack) {
+        workspace.backpack.close();
+      }
+    }
+  };
+
+  /**
+   * Update the cursor (and possibly the trash can lid) to reflect whether the
+   * dragging block would be deleted if released immediately.
+   * @private
+   */
+  Blockly.BlockDragger.prototype.updateCursorDuringBlockDrag_ = function() {
+    this.wouldDeleteBlock_ = this.draggedConnectionManager_.wouldDeleteBlock();
+    const trashcan = this.workspace_.trashcan;
+    if (this.wouldDeleteBlock_) {
+      this.draggingBlock_.setDeleteStyle(true);
+      if (this.deleteArea_ == Blockly.DELETE_AREA_TRASH && trashcan) {
+        trashcan.setLidOpen(true);
+      }
+    } else {
+      this.draggingBlock_.setDeleteStyle(false);
+      if (trashcan) {
+        trashcan.setLidOpen(false);
+      }
+    }
```

Comment:
I see this as being a change that should happen in the Base `BlockDragger`. If it was extended as a custom Dragger, then multiple plugins that use this feature would not be compatible and I think that it would be common for plugins to want to know when a cursor enters/exits over it (when holding block).